tests/CompletionEngineTests/Manipulator/XmlParserTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using Avalonia.Ide.CompletionEngine;
+using Xunit;
+
+namespace CompletionEngineTests.Manipulator
+{
+    /// <summary>
+    /// Tests for XmlParser behavior on which TextManipulator is dependent
+    /// </summary>
+    public class XmlParserTests
+    {
+        [Fact]
+        public void Should_BeInNoneState_When_OnClosingBrace()
+        {
+            var parser = XmlParser.Parse("<Grid>");
+            Assert.Equal(XmlParser.ParserState.None, parser.State);
+        }
+
+        [Fact]
+        public void Should_NotBeInClosingTag_When_StartTag()
```

Comment:
@kekekeks are you ok with such test names? I thought they need to be more strict to better picture what is being tested in this file.

Reply:
No particular preference on test naming as long as names are readable.